src/UI/Components/RelatedInventoriesProvider/RelatedInventoriesProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useContext } from "react";
+import { ServiceModel } from "@/Core";
+import {
+  Inventories,
+  useGetRelatedInventories,
+} from "@/Data/Managers/V2/GetRelatedInventories";
+import { DependencyContext, words } from "@/UI";
+import { ErrorView, LoadingView } from "@/UI/Components";
+import { findInterServiceRelations } from "../Diagram/helpers";
+
+interface Props {
+  serviceModels: ServiceModel[];
+  mainService: ServiceModel;
+  Dependant: React.FC<{
+    services: ServiceModel[];
+    mainService: ServiceModel;
+    relatedInventories: Inventories;
+  }>;
+}
+
+/**
+ * Renders the RelatedInventoriesProvider component.
+ * It serves purpose to extract the data fetching logic and provide the required service inventories data for the Dependent
+ *
+ * @param {ServiceModel[]} ServiceModels - The Array of all Service Models.
+ * @param {ServiceModel} mainService - The main service model used by Dependant.
+ * @param {React.FC} Dependant - The Component that depends it's lifecycle on the instance data.
+ * @returns {JSX.Element} The rendered RelatedInventoriesProvider component.
+ */
+export const RelatedInventoriesProvider: React.FC<Props> = ({
+  serviceModels,
+  mainService,
+  Dependant,
+}) => {
+  const { environmentHandler } = useContext(DependencyContext);
+  const environment = environmentHandler.useId();
+
+  const relatedCatalogsNames = findInterServiceRelations(mainService);
```

Comment:
`relatedCatalogsNames` is not right, we only have on catalog, so I'd go for `relatedInterServiceRelationNames ` or `relatedInterServiceRelation` whichever matches best. Also can't hurt to add typing to the variable, to indicate it's an array of strings, or specific items.